### ISSUE TITLE
Allow Custom Components to be uploaded with prefabs and scenes

### ISSUE THREAD

**Issue by Nolankicks** [Facepunch/sbox-issues#5802]
### For?

S&Box

### What can't you do?

I’m unable to upload custom components with my prefabs and scenes.

### How would you like it to work?

I would like it to work by toggling the upload source files would upload the components. Or a separate upload components toggle would also be nice.

### What have you tried?

They are marked as missing. They load fine when opening the scene. But I would like to expose some values for my components as prefab variables for ugc. They don’t even have to be completely uploaded. Just the properties would be good. So the properties could be easily modified.

**Comment by KingOfTheFBI** [comment]
Any update on this? With maps being uploaded as scenes and soon map scenes this is going to be pretty important. I have a map that has a few custom components that are needed for it to function properly.

**Comment by Nolankicks** [comment]
@KingOfTheFBI this kinda exists already if you use TTT components in it. They will work as TTT components when you play the map.

**Comment by KingOfTheFBI** [comment]
> [@KingOfTheFBI](https://github.com/KingOfTheFBI) this kinda exists already if you use TTT components in it. They will work as TTT components when you play the map.

Yeah I meant the custom components working for the scene when you load it for anything, even if they aren't part of the game you're loading. Idk if they want that functionality though, but it would be really helpful. Maybe the game can choose to block it?

An example would be making a custom logic system to open a part of the map instead of having the gamemode provide the functionality. Another example if you wanted to add something like a working pinball machine players can use

**Comment by Retroeer** [comment]
what functionality is your map relying on that can't be solved with actiongraph?

**Comment by KingOfTheFBI** [comment]
> what functionality is your map relying on that can't be solved with actiongraph?

It's a custom component written in code. Shouldn't just be action graph you can use
